assets/js/admin/marketplace-suggestions.js

```diff
@@ -0,0 +1,411 @@
+					'_wpnonce': marketplace_suggestions.dismiss_suggestion_nonce,
+					'slug': suggestionSlug,
+				}
+			);
+
+			window.wcTracks.recordEvent( 'marketplace_suggestion_dismissed', {
+				suggestionSlug: suggestionSlug
+			} );
+		}
+
+		// Render DOM element for suggestion dismiss button.
+		function renderDismissButton( suggestionSlug ) {
+			var dismissButton = document.createElement( 'a' );
+
+			dismissButton.classList.add( 'suggestion-dismiss' );
+			dismissButton.setAttribute( 'href', '#' );
+			dismissButton.onclick = function( event ) {
+				event.preventDefault();
+				dismissSuggestion( suggestionSlug );
+			}
+
+			return dismissButton;
+		}
+
+		function addUTMParameters( context, url ) {
+			var utmParams = {
+				utm_source: 'unknown',
+				utm_campaign: 'marketplacesuggestions',
+				utm_medium: 'product'
+			};
+			var sourceContextMap = {
+				'productstable': [
+					'products-list-inline'
+				],
+				'productsempty': [
+					'products-list-empty-header',
+					'products-list-empty-footer',
+					'products-list-empty-body'
+				],
+				'ordersempty': [
+					'orders-list-empty-header',
+					'orders-list-empty-footer',
+					'orders-list-empty-body'
+				],
+				'editproduct': [
+					'product-edit-meta-tab-header',
+					'product-edit-meta-tab-footer',
+					'product-edit-meta-tab-body'
+				]
+			};
+			var utmSource = _.findKey( sourceContextMap, function( sourceInfo ) {
+				return _.contains( sourceInfo, context );
+			} );
+			if ( utmSource ) {
+				utmParams.utm_source = utmSource;
+			}
+
+			return url + '?' + jQuery.param( utmParams );
+		}
+
+		// Render DOM element for suggestion linkout, optionally with button style.
+		function renderLinkout( context, slug, url, text, isButton ) {
+			var linkoutButton = document.createElement( 'a' );
+
+			var utmUrl = addUTMParameters( context, url );
+			linkoutButton.setAttribute( 'href', utmUrl );
+			linkoutButton.setAttribute( 'target', 'blank' );
+			linkoutButton.textContent = text;
+
+			linkoutButton.onclick = function( event ) {
+				window.wcTracks.recordEvent( 'marketplace_suggestion_clicked', {
+					suggestionSlug: slug
+				} );
+			}
+
+			if ( isButton ) {
+				linkoutButton.classList.add( 'button' );
+			}
+			else {
+				linkoutButton.classList.add( 'linkout' );
+				var linkoutIcon = document.createElement( 'span' );
+				linkoutIcon.classList.add( 'dashicons', 'dashicons-external' );
+				linkoutButton.appendChild( linkoutIcon );
+			}
+
+			return linkoutButton;
+		}
+
+		// Render DOM element for suggestion icon image.
+		function renderSuggestionIcon( iconUrl ) {
+			if ( ! iconUrl ) {
+				return null;
+			}
+
+			var image = document.createElement( 'img' );
+			image.src = iconUrl;
+			image.classList.add( 'marketplace-suggestion-icon' );
+
+			return image;
+		}
+
+		// Render DOM elements for suggestion content.
+		function renderSuggestionContent( title, copy ) {
+			var left = document.createElement( 'div' );
```

Comment:
Super minor but in this method and `renderSuggestionCTA` the var names of `left` and `right` which I'm guessing align with how they are ultimately displayed on the page seem a bit odd in the methods. I suppose something like `container` or `wrapper` might make more sense - kind of like how you call it a `row` in `renderTableBanner` below. Again super minor just made me pause when reading through the code.

Reply:
The perils of lazy refactoring. Will look at fixing this as long as I'm confident it doesn't destabilise things :)